intake/source/cache.py

```diff
@@ -238,6 +241,55 @@ def _download(file_in, file_out, blocksize, output=False):
         display.remove(out)
 
 
+class FileCache(BaseCache):
+    def _make_files(self, urlpath):
+        from dask.bytes import open_files
+
+        self._ensure_cache_dir()
+        subdir = self._hash(urlpath)
+        files_in = open_files(urlpath, 'rb')
+        files_out = [open_files([self._path(f.path, subdir)], 'wb')[0]
+                     for f in files_in]
+        return files_in, files_out
+
+
+class DirCache(BaseCache):
+    def _make_files(self, urlpath, **kwargs):
+        from dask.bytes import open_files
+
+        self._ensure_cache_dir()
+        subdir = self._hash(urlpath)
+        depth = self._spec['depth']
+        files_in = []
+        for i in range(1, depth + 1):
+            files_in.extend(open_files('/'.join([urlpath] + ['*']*i)))
+        files_out = [open_files([self._path(f.path, subdir)], 'wb')[0]
+                     for f in files_in]
+        files_in2, files_out2 = [], []
+        paths = set(os.path.dirname(f.path) for f in files_in)
+        print(paths)
+        for fin, fout in zip(files_in, files_out):
+            print(fin, fout)
+            if fin.path in paths:
+                try:
+                    print('mkdir', fout.path)
+                    os.makedirs(fout.path)
+                except Exception:
+                    pass
+            else:
+                files_in2.append(fin)
+                files_out2.append(fout)
+        return files_in2, files_out2
+
+
+class CompressedCache(BaseCache):
+    def _make_files(self, urlpath, **kwargs):
+        import dask.bytes.compression
+        import tempfile
+        ext = os.path.splitext(urlpath)[1]
+        comp = kwargs.get('compression', ext)
```

Comment:
Maybe inherit from `FileCache` so you can unpack then call its `_make_files`?

Reply:
I haven't figured out what to do yet. The dir-cache is maybe ready, but needs tests, of course.